commands/box/contract/contract.go

```diff
@@ -1247,3 +1220,128 @@ func detailAbi(cmd *cobra.Command, args []string) {
 		fmt.Printf("  %s:  %v(%v)\n", e.ID(), e.Name, strings.Join(inputs, ", "))
 	}
 }
+
+func getIndex(v interface{}) (int, error) {
+	var (
+		max              int
+		index            int
+		iFiles           []int
+		srcFile          string
+		originalContract bool
+	)
+	srcFile, ok := v.(string)
+	if ok {
+		originalContract = false
+	} else {
+		originalContract, ok = v.(bool)
+		if !ok {
+			return -1, errors.New("please check your parameter")
+		}
+	}
+	if err := util.MkDir(abiDir); err != nil {
+		panic(err)
+	}
+	if originalContract {
+		if _, err := findOriginalContract(); err != nil {
+			fmt.Println()
+			return -1, err
+		}
+		index = 0
+	} else {
+		if err := util.FileExists(srcFile); err != nil {
+			return -1, err
+		}
+		err := filepath.Walk(abiDir, func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			s := strings.TrimPrefix(path, abiDir)
+			if i, err := strconv.Atoi(s); err == nil && i >= 0 {
+				iFiles = append(iFiles, i)
+			}
+			return nil
+		})
+		if err != nil {
+			panic(err)
+		}
+		if len(iFiles) > 0 {
+			for _, v := range iFiles {
+				if max < v {
+					max = v
+				}
+			}
+		}
+		index = max + 1
+	}
+	return index, nil
+}
+
+func writeAbi(srcFile string, description string, index int) error {
+	newFile := abiDir + strconv.Itoa(index)
+	data, err := ioutil.ReadFile(srcFile)
+	if err != nil {
+		panic(err)
+	}
+	if _, err := abi.JSON(bytes.NewBuffer(data)); err != nil {
+		return errors.New("illegal abi file")
+	}
+	// Write data to desc
+	err = ioutil.WriteFile(newFile, data, 0644)
+	if err != nil {
+		panic(err)
+	}
+	f, err := os.OpenFile(recordFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+	note := fmt.Sprintf("%d%s%s, %s%s%s\n", index, abiSep, description,
+		time.Now().Format(time.Stamp), abiSep, newFile)
+	if _, err := f.WriteString(note); err != nil {
+		panic(err)
+	}
+	return nil
+}
+
+func findOriginalContract() (string, error) {
```

Comment:
var possiblePlace []string  (home dir, contracts/bonus.abi, .cmd/contract/test/bonus.abi)
for _, p := range possiblePlace {

}

Reply:
on the other side, think about how user feel easy to update original genesis contract abi

Reply:
now user cannot correctly update genesis contract abi